include/dlaf/common/index2d.h

```diff
@@ -75,19 +77,17 @@ class basic_coords {
     std::swap(row_, col_);
   }
 
-  /// Given a coordinate, returns its transposed with the same type.
-  template <class Coords2DType>
-  friend Coords2DType transposed(const Coords2DType& coords);
-
   /// Adds "(<row_>, <col_>)" to out.
   friend std::ostream& operator<<(std::ostream& out, const basic_coords& index) {
-    if (std::is_same<IndexT, signed char>::value || std::is_same<IndexT, char>::value) {
+    if (std::is_same<IndexT, signed char>::value || std::is_same<IndexT, char>::value)
       return out << "(" << static_cast<int>(index.row_) << ", " << static_cast<int>(index.col_) << ")";
-    }
     return out << "(" << index.row_ << ", " << index.col_ << ")";
   }
 
 protected:
+  // NOTE: operator== and operator! are protected otherwise it would be possible to compare Index2D and
+  // Size2D or same type but mixing IndexType and/or Tag. Which is something not desired.
```

Comment:
As it is a template argument you cannot mix the IndexType.

Reply:
You're right.

```suggestion
  // Size2D or same type but mixing Tag. Which is something not desired.
```